contracts/crowdfund/AuctionCrowdfundBase.sol

```diff
@@ -302,4 +302,36 @@ abstract contract AuctionCrowdfundBase is Crowdfund {
             revert InvalidAuctionIdError();
         }
     }
+
+    function _finalize(
+        CrowdfundLifecycle lc,
+        IMarketWrapper market_,
+        uint256 auctionId_,
+        uint96 lastBid_
+    ) internal {
+        // Mark as busy to prevent `burn()`, `bid()`, and `contribute()`
+        // getting called because this will result in a `CrowdfundLifecycle.Busy`.
+        _bidStatus = AuctionCrowdfundStatus.Busy;
+        {
+            // If we've bid before or the CF is not expired,
+            // finalize the auction.
+            if (lastBid_ != 0 || lc == CrowdfundLifecycle.Active) {
+                if (!market_.isFinalized(auctionId_)) {
+                    (bool s, bytes memory r) = address(market_).call(
+                        abi.encodeCall(IMarketWrapper.finalize, auctionId_)
+                    );
+                    if (!s) {
+                        // Allow call to finalize the auction to fail if the
+                        // crowdfund has expired and we are not the highest bidder.
+                        if (
+                            lc != CrowdfundLifecycle.Expired ||
+                            market_.getCurrentHighestBidder(auctionId_) == address(this)
+                        ) {
+                            r.rawRevert();
```

Comment:
Is there a reason to try-fail-allow here, or would it be simpler to skip calling the market wrapper at all if we know the crowdfund is expired and we are not the highest bidder?

Reply:
If the crowdfund was the highest bidder, it would win the auction even though the crowdfund expired. In this case, we'd want to finalize and declare a win if the crowdfund owns the NFT instead of expiring.

Reply:
Yeah that would make more sense.

Would adding this condition to exit `_finalize()` early if the crowdfund has expired and is not the highest bidder work?

```solidity
if (
    lc == CrowdfundLifecycle.Expired &&
    market_.getCurrentHighestBidder(auctionId_) != address(this)
) {
    return;
}
```